File: Special Array I.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        int n = nums.size();
        int parity = nums[0] & 1;
        for (int i = 1; i < n; i++) {
            int new_parity = nums[i] & 1;
            if (parity == new_parity) return false;
            parity = new_parity;
        }
        return true;
    }
};

File: Special Array I.go
func isArraySpecial(nums []int) bool {
    n := len(nums)
    parity := nums[0] & 1;
    for i := 1; i < n; i++ {
        new_parity := nums[i] & 1
        if parity == new_parity {
            return false
        }
        parity = new_parity
    }
    return true
}
